### DISCUSSION TITLE
Is The First Execution Slow Always?

### DISCUSSION THREAD

**Discussion by Yuji-github** [google/flax#2383] [Q&A] [answered]
I tested the Binary Image classification model on Google Colab by using Flax. 
https://github.com/Yuji-github/Jax_ResNet18_Classification

My first execution (running the cell for the first time) is always prolonged. 
Then, the second execution (re-running the cell) is quicker. 

Is that because Flax carries all of the data on GPU before the first execution?

**Comment by jheek** [comment, marked as answer]
Ops and compiled functions (annotated with jax.jit, jax.pmap, ...) are compiled the first time you call them. This is what normally causes the slowdown.
